doc/man3/ASYNC_WAIT_CTX_new.pod

```diff
@@ -103,14 +120,56 @@ code that the job should be resumed by making the wait file descriptor
 "readable". Once resumed the engine should clear the wake signal on the wait
 file descriptor.
 
+As well as a file descriptor, user code may also be notified via a callback. The
+callback and data pointers are stored within the ASYNC_WAIT_CTX along with an
+additional status field that can be used for the notification of retries from an
+engine. This additional method can be used when the user thinks the a file
+descriptor is too costly in terms of CPU cycles or in some context where a file
+descriptor is not appropriate.
+
+ASYNC_WAIT_CTX_set_callback() sets the callback and the callback argument. The
+callback will be called to notify user code when an engine completes a
+cryptography operation.
+
+ASYNC_WAIT_CTX_get_callback() returns the callback set in the ASYNC_WAIT_CTX
+structure.
+
+ASYNC_WAIT_CTX_set_status() allows an engine to set the current engine status.
+The possible status values are the following:
+ASYNC_STATUS_UNSUPPORTED: The engine does not support the callback mechanism.
+This is the default value.
+ASYNC_STATUS_ERR: The engine has a fatal problem with this request. The user
+code should clean up this session.
+ASYNC_STATUS_OK: The request has been successfully submitted.
+ASYNC_STATUS_EAGAIN: The engine has some problem which will be recovered soon,
+such as buffer is full, so user code should send the request again next time.
+The engine must call ASYNC_WAIT_CTX_set_status() to set the status to some value
+other than ASYNC_STATUS_UNSUPPORTED if it intends to later invoke the callback.
+
+ASYNC_WAIT_CTX_get_status() allows user code to obtain the current status value.
+If the status is any value other than ASYNC_STATUS_OK then the user code should
+not expect to receive a callback from the engine even if one has been set.
```

Comment:
These two paragraphs seem to contradict each other. The first says that engine should set the value to something other than ASYNC_STATUS_UNSUPPORTED if it plans to call the callback. But the second one says that the callback will only be called if the status is ASYNC_STATUS_OK. So it is unclear whether the application should expect a callback if the status is ASYNC_STATUS_EAGAIN (for example).

Reply:
It is not contradict. Set value other than ASYNC_STATUS_UNSUPPORTED, means the callback mechanism instead of "fd" is used. "ASYNC_STATUS_OK" means that the callback will be triggered in the engine part, and if not, such as "ASYNC_STATUS_EAGAIN", then user code need to call the callback itself. 
I have re-word the sentence to make it more clear.